impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -7,30 +7,36 @@
 #include <memory>
 
 #include "flutter/fml/memory/ref_ptr.h"
-#include "flutter/vulkan/procs/vulkan_handle.h"
-#include "flutter/vulkan/procs/vulkan_proc_table.h"
 #include "impeller/renderer/backend/vulkan/device_buffer_vk.h"
 #include "impeller/renderer/backend/vulkan/formats_vk.h"
 #include "impeller/renderer/backend/vulkan/texture_vk.h"
 #include "impeller/renderer/formats.h"
 
 namespace impeller {
 
-AllocatorVK::AllocatorVK(ContextVK& context,
+AllocatorVK::AllocatorVK(std::weak_ptr<Context> context,
                          uint32_t vulkan_api_version,
                          const vk::PhysicalDevice& physical_device,
                          const vk::Device& logical_device,
                          const vk::Instance& instance,
                          PFN_vkGetInstanceProcAddr get_instance_proc_address,
                          PFN_vkGetDeviceProcAddr get_device_proc_address)
-    : context_(context), device_(logical_device) {
+    : context_(std::move(context)), device_(logical_device) {
   vk_ = fml::MakeRefCounted<vulkan::VulkanProcTable>(get_instance_proc_address);
 
   auto instance_handle = vulkan::VulkanHandle<VkInstance>(instance);
-  FML_CHECK(vk_->SetupInstanceProcAddresses(instance_handle));
+  if (!vk_->SetupInstanceProcAddresses(instance_handle)) {
```

Comment:
curious, why not `FML_CHECK` and fail early like before?

Reply:
The allocator, like most other stuff is setup during context creation. One of the things we'd eventually want is to attempt to setup a Vulkan context on the device first and then fallback to GL. I want to minimize the number of things that are considered fatal in attempting the Vulkan setup since that would make it impossible to fallback to anything else. Having said that, this make be a case where the check was fine. I was just cleaning up checks with a view to also setup the fallback.